kernel/include/kswap.h

```diff
@@ -62,6 +64,7 @@ static ALWAYS_INLINE unsigned int do_swap(unsigned int key,
 					  int is_spinlock)
 {
 	ARG_UNUSED(lock);
+	k_spinlock_key_t schedkey = {};
```

Comment:
Why initialized here? /nit

In my experience initializing variables when they are expected to be overwritten later is an antipattern and tends to lead to even stranger bugs down the line.

(I'm used to late-declaration, which sidesteps this problem in many cases, not to mention often producing better code... but Zephyr's coding style doesn't do that.)

Reply:
The compiler isn't always very good at detecting that all paths lead to assignment.  Honestly I don't remember if this was true here or if the initializer just left my fingers unbidden.

Reply:
I know what you mean - I hit cases of

```
foo_t *f;
if (complex_expression_a) {
    f = bar();
}
...
if (complex_expression_b) {
    baz(f);
}
```
where it's nontrivial to determine if complex_expression_b actually implies complex_expression_a, all the time. (e.g. if the expressions are both `qux->some_flag`, and the compiler didn't manage to determine that nothing aliased with `some_flag` or `qux`).

...though I'd be mildly surprised if the compiler couldn't detect it in this particular case. There's an unconditional assignment down at line 78.